src/crimson/os/seastore/seastore_types.h

```diff
@@ -857,11 +857,14 @@ constexpr auto PLACEMENT_HINT_NULL = placement_hint_t::NUM_HINTS;
 
 std::ostream& operator<<(std::ostream& out, placement_hint_t h);
 
-enum alignas(4) device_type_t : uint_fast8_t {
+enum class alignas(4) device_type_t : uint_fast8_t {
   NONE = 0,
-  SEGMENTED, // i.e. Hard_Disk, SATA_SSD, NAND_NVME
-  RANDOM_BLOCK, // i.e. RANDOM_BD
-  PMEM, // i.e. NVDIMM, PMEM
+  HDD,
+  SSD,
+  ZNS,
+  RANDOM_BLOCK_SSD,
+  SEGMENTED_EPHEMERAL,
+  RANDOM_BLOCK_EPHEMERAL,
```

Comment:
Probably worth reorder to group the same backend type together, make it easier for `get_default_backend_of_device()`

Reply:
How about
```cpp
enum class alignas(4) device_type_t : uint_fast8_t {
  NONE = 0,
  HDD,
  SSD,
  ZNS,
  SEGMENTED_EPHEMERAL,
  RANDOM_BLOCK_SSD,
  RANDOM_BLOCK_EPHEMERAL,
  NUM_TYPES
};
constexpr backend_type_t get_default_backend_of_device(device_type_t dtype) {
  assert(dtype != device_type_t::NONE &&
         dtype != device_type_t::NUM_TYPES);
  if (dtype >= device_type_t::HDD &&
      dtype <= device_type_t::SEGMENTED_EPHEMERAL) {
    return backend_type_t::SEGMENTED;
  } else {
    return backend_type_t::RANDOM_BLOCK;
  }
}
```